src/Membership/MembershipRequestHandler.php

```diff
@@ -52,6 +55,7 @@ public function __construct(
         $this->mailer = $mailer;
         $this->manager = $manager;
         $this->referentTagManager = $referentTagManager;
+        $this->referentZoneManager = $referentZoneManager;
```

Comment:
what about creating a listener for `UserEvents::USER_UPDATED` event instead and moving this logic into?

Reply:
As discussed in PM, for now we left it like it is.